src/Features/Core/Portable/CodeRefactorings/SyncNamespace/AbstractSyncNamespaceService.cs

```diff
@@ -0,0 +1,145 @@
+            Document document, int position, CancellationToken cancellationToken);
+
+        protected SyntaxAnnotation WarningAnnotation
+            => CodeActions.WarningAnnotation.Create(
+                FeaturesResources.Warning_colon_changing_namespace_may_produce_invalid_code_and_change_code_meaning);
+
+        protected async Task<bool> ContainsPartialTypeWithMultipleDeclarationsAsync(
+            Document document, SyntaxNode compilationUnitOrNamespaceDecl, CancellationToken cancellationToken)
+        {
+            var memberDecls = GetMemberDeclarationsInContainer(compilationUnitOrNamespaceDecl);
+            var semanticModel = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+            var semanticFacts = document.GetLanguageService<ISemanticFactsService>();
+
+            foreach (var memberDecl in memberDecls)
+            {
+                var memberSymbol = semanticModel.GetDeclaredSymbol(memberDecl, cancellationToken);
+
+                // Simplify the check by assuming no multiple partial declarations in one document
+                if (memberSymbol is ITypeSymbol typeSymbol
+                    && typeSymbol.DeclaringSyntaxReferences.Length > 1
+                    && semanticFacts.IsPartial(typeSymbol, cancellationToken))
+                {
+                    return true;
+                }
+            }
+            return false;
+        }
+
+        /// <summary>
+        /// Try get the relative namespace for <paramref name="namespace"/> based on <paramref name="relativeTo"/>,
+        /// if <paramref name="relativeTo"/> is the containing namespace of <paramref name="namespace"/>.
+        /// For example:
+        /// - If <paramref name="relativeTo"/> is "A.B" and <paramref name="namespace"/> is "A.B.C.D", then
+        /// the relative namespace is "C.D".
+        /// - If <paramref name="relativeTo"/> is "A.B" and <paramref name="namespace"/> is also "A.B", then
+        /// the relative namespace is "".
+        /// - If <paramref name="relativeTo"/> is "" then the relative namespace us <paramref name="namespace"/>.
+        /// </summary>
+        private static string GetRelativeNamespace(string relativeTo, string @namespace, ISyntaxFactsService syntaxFacts)
+        {
+            Debug.Assert(relativeTo != null && @namespace != null);
+
+            if (syntaxFacts.StringComparer.Equals(@namespace, relativeTo))
+            {
+                return string.Empty;
+            }
+            else if (relativeTo.Length == 0)
+            {
+                return @namespace;
+            }
+
+            var containingText = relativeTo + ".";
+            if (syntaxFacts.IsCaseSensitive
+                ? @namespace.StartsWith(containingText, StringComparison.Ordinal)       // For C#
+                : CaseInsensitiveComparison.StartsWith(@namespace, containingText))     // for VB
```

Comment:
ternary *in* conditional is super hard to read.  please extract.

Reply:
syntaxFacts gives you a StringComparer you can use that is correct per language.

Reply:
Fixed